repository-creation/create_repository.go

```diff
@@ -0,0 +1,240 @@
+	fmt.Println("This script will create a new ONS Digital Publishing repository." +
+		"In order to create and configure a new repository please answer the prompts.")
+
+	accessToken, userHandle, repoName, repoDescription, defaultBranch := getConfigurationsForNewRepo(name, ProjectType, personalAccessToken, isLibrary)
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: accessToken},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	client := github.NewClient(tc)
+
+	hasAccess, err := checkAccess(ctx, client)
+	if !hasAccess {
+		log.Event(ctx, "user does not have access", log.Error(err))
+		return cloneUrl, err
+	}
+
+	repo := &github.Repository{
+		Name:          github.String(repoName),
+		Description:   github.String(repoDescription),
+		DefaultBranch: github.String(defaultBranch),
+		MasterBranch:  github.String(masterBranch),
+		Private:       github.Bool(false),
+		HasWiki:       github.Bool(false),
+		HasProjects:   github.Bool(false),
+		AutoInit:      github.Bool(true),
+	}
+
+	err = createRepo(ctx, client, repo)
+	if err != nil {
+		log.Event(ctx, "unable to create repository", log.Error(err))
+		return cloneUrl, err
+	}
+
+	err = createDevelopBranch(ctx, client, repoName)
+	if err != nil {
+		log.Event(ctx, "unable to create develop branch", log.Error(err))
+		return cloneUrl, err
+	}
+
+	err = setDevelopAsDefaultBranch(ctx, client, repoName, repo)
+	if err != nil {
+		log.Event(ctx, "failed to set default branch to develop", log.Error(err))
+		return cloneUrl, err
+	}
+
+	err = setBranchProtections(ctx, client, repoName)
+	if err != nil {
+		log.Event(ctx, "unable to set all branch protections, please review and correct these manually", log.Error(err))
+		return cloneUrl, err
+	}
+
+	err = setTeamsAndCollaborators(ctx, client, repoName, userHandle)
+	if err != nil {
+		log.Event(ctx, "unable to set team and collaborators", log.Error(err))
+		return cloneUrl, err
+	}
+	repositoryObj, _, err := client.Repositories.Get(ctx, org, repoName)
+	if err != nil {
+		log.Event(ctx, "unable to locate the the attempted newly created repository", log.Error(err))
+		return cloneUrl, err
+	}
+	cloneUrl = repositoryObj.GetCloneURL()
+	// Notify user of completion and get them to turn off actions
+	log.Event(ctx, "repository has successfully been create please Disable Actions for this repository")
+	return cloneUrl, nil
+}
+
+// setTeamsAndCollaborators will set the DigitalPublishing team as a team working on the repo and removes the creator from being a collaborator
+func setTeamsAndCollaborators(ctx context.Context, client *github.Client, repoName string, userHandle string) error {
+	addTeamRepoOptions := github.TeamAddTeamRepoOptions{Permission: "admin"}
+	resp, err := client.Teams.AddTeamRepo(ctx, teamID, org, repoName, &addTeamRepoOptions)
+	if err != nil {
+		log.Event(ctx, "unable to add collaborators", log.Error(err))
```

Comment:
if we can't add collaborators, we probably should return here so we dont try and remove the only name left?